### PR TITLE
[Ttahub-1156] Continued Performance issues when multiple recipients are selected on an AR

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
Multiple methods of increasing performance through better use of sequelize 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1156
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1169

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — APPROVED** [review]
I'm not seeing huge performance gains vs main-ar-redesign locally but all these changes make sense given your expressed findings. Everything appears to being created in the DB as expected. Will wait for someone to put in an additional review before merging though

File: src/services/goals.js
Comment (kryswisnaskas) [inline src/services/goals.js:229]: Is there a possibility that objective here would be null? If so, this will throw the undefined error.  (assuming that's the desired behavior)

File: src/services/goals.js
Comment (kryswisnaskas) [inline src/services/goals.js:246]: `o `- same comment as line 229

**Review by kryswisnaskas — COMMENTED** [review]
👍 
I wasn't able to fully review this, but submitting an activity report was verified. I left couple of comments for now. 
We want to merge this before a testing session.

**Pull request merged**
